tests/tape/test_qscript.py

```diff
@@ -292,6 +293,27 @@ def test_update_output_dim(self, m, output_dim, ops, factor):
         qs = QuantumScript(ops, m)
         assert qs.output_dim == output_dim * factor
 
+    def test_lazy_batch_size_and_output_dim(self):
+        """Test that batch_size and output_dim are computed lazily."""
+        qs = QuantumScript([qml.RX([1.1, 2.2], 0)], [qml.expval(qml.PauliZ(0))])
+        copied = qs.copy()
+        assert qs._batch_size is _UNSET_BATCH_SIZE
+        assert qs._output_dim is None
+        # copying did not evaluate them either
+        assert copied._batch_size is _UNSET_BATCH_SIZE
+        assert copied._output_dim is None
+
+        # now evaluate it
+        assert qs.batch_size == 2
+        assert qs._output_dim is None  # setting batch_size didn't set output_dim
+        assert qs.output_dim == 2
+        copied = qs.copy()
+        assert qs._batch_size == 2
+        assert qs._output_dim == 2
+        # copied tape has it pre-evaluated
+        assert copied._batch_size == 2
+        assert copied._output_dim == 2
```

Comment:
Shall we add a test that is sensitive to whether or not `_update_output_dim` sets `_batch_size`? Because we're discussing this above and it is definitely not the most stable setup...

Reply:
will do!

Reply:
done 😎